File: amazia-console/app/Delivery/Controller/ListDeliveryController.php
<?php

namespace App\Delivery\Controller;

use App\Delivery\Service\ListDeliveryService;
use App\Http\Controllers\Controller;
use Illuminate\Http\Request;

/**
 * 配送一覧 Controller（GET /api/deliveries[?shippingStatusId=N]）。
 */
class ListDeliveryController extends Controller
{
    private ListDeliveryService $service;

    public function __construct(ListDeliveryService $service)
    {
        $this->service = $service;
    }

    public function __invoke(Request $request)
    {
        $statusId = $request->query('shippingStatusId');
        $response = $this->service->list($statusId !== null ? (int) $statusId : null);
        return response()->json($response->json(), $response->status());
    }
}

File: amazia-core/src/main/java/com/example/delivery/controller/ListDeliveryController.java
package com.example.delivery.controller;

import com.example.delivery.dto.DeliveryResponse;
import com.example.delivery.service.ListDeliveryService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**
 * 配送一覧 API（GET /api/deliveries[?shippingStatusId=N]）。
 */
@RestController
@RequestMapping("/api")
public class ListDeliveryController {

    private final ListDeliveryService service;

    public ListDeliveryController(ListDeliveryService service) {
        this.service = service;
    }

    @GetMapping("/deliveries")
    public ResponseEntity<List<DeliveryResponse>> list(
            @RequestParam(required = false) Long shippingStatusId) {
        return ResponseEntity.ok(service.list(shippingStatusId));
    }
}
